pkg/rbd/nodeserver.go

```diff
@@ -386,7 +394,7 @@ func (ns *NodeServer) mountVolumeToStagePath(ctx context.Context, req *csi.NodeS
 	existingFormat, err := diskMounter.GetDiskFormat(devicePath)
 	if err != nil {
 		klog.Errorf(util.Log(ctx, "failed to get disk format for path %s, error: %v"), devicePath, err)
-		return err
+		return readOnly, err
 	}
 
 	if existingFormat == "" && !staticVol {
```

Comment:
Do not attempt to create a filesystem on a ROX volume

Reply:
Thanks for the pointers @dillaman, will make necessary changes

Reply:
@dillaman if I don't format it am hitting `cannot mount unformatted disk %s as we are manipulating it in read-only mode` https://github.com/kubernetes/utils/blob/master/mount/mount_linux.go#L353

Reply:
Correct. ROX would need to be a pre-existing disk image (a la static PV) -- or you would need to move all the formatting logic to the `ControllerCreateVolume` call to avoid the potential for races between nodes attempting to format the disk (ROX must always be read-only).

Reply:
one use case where the ROX need not be static PV is, cloning a PVC and mounting it as readOnly(ROX).

in `ControllerCreateVolume` we are supposed to create rbd image and map the rbd image and format it ext4 or xfs based on the format type and unmap it one the formatting is completed isn't it?

Reply:
> one use case where the ROX need not be static PV is, cloning a PVC and mounting it as readOnly(ROX).

Cloning/create from snapshot wouldn't be an issue since you would already have a file system on the RBD image.

> in `ControllerCreateVolume` we are supposed to create rbd image and map the rbd image and format it ext4 or xfs based on the format type and unmap it one the formatting is completed isn't it?

`ControllerCreateVolume` just creates the image, correct? I think all mapping and formatting is deferred to the node server plug-in.

Reply:
yes, in the current code base mapping and formatting are deferred to the node server.

Reply:
I'm not really suggesting to move it, I am just saying that would be the only thing you could do to safely allow the creation of file systems on ROX volumes. However, I think the ROX use-case should be for static PVs or PVs from existing PVs/snapshots, so no need to re-engineer the code.

Reply:
okay let me do some more testing and get back to you :)